pkg/ingester/ingester_v2.go

```diff
@@ -412,10 +434,16 @@ func NewV2(cfg Config, clientConfig client.Config, limits *validation.Overrides,
 	// them from the underlying system (ie. TSDB).
 	if registerer != nil {
 		registerer.Unregister(i.metrics.memSeries)
+
 		promauto.With(registerer).NewGaugeFunc(prometheus.GaugeOpts{
 			Name: "cortex_ingester_memory_series",
 			Help: "The current number of series in memory.",
-		}, i.numSeriesInTSDB)
+		}, i.getMemorySeriesMetric)
+
+		promauto.With(registerer).NewGaugeFunc(prometheus.GaugeOpts{
```

Comment:
I'm wondering if it makes sense to include oldest sample as well, but I guess it doesn't give us anything extra.

Reply:
I'm not sure which value would give exposing the oldest sample (I guess you mean the oldest block `mint`). I believe that from the user's perspective is easier to reason about the time the block has been compacted so that this metric effectively measure the delay between it and when it was successfully shipped to the storage.

Reply:
> I guess you mean the oldest block `mint`

Yes. It was just an idea said out loud while thinking if block timestamp is the right thing to expose, but I don't see much value in it.